iceoryx_posh/include/iceoryx_posh/roudi/iceoryx_roudi_components.hpp

```diff
@@ -39,25 +44,12 @@ struct IceOryxRouDiComponents
     /// @brief Handles MemoryProvider and MemoryBlocks
     IceOryxRouDiMemoryManager m_rouDiMemoryManager;
 
-    /// @todo Make this an object
-    /// @brief Cleanup resources and initalize memory
-    cxx::GenericRAII m_initalizer{
-        [this]() {
-            // this temporary object will create a roudi mqueue and close it immediatelly
-            // if there was an outdated roudi message queue, it will be cleaned up
-            // if there is an outdated mqueue, the start of the apps will be terminated
-            runtime::MqBase::cleanupOutdatedMessageQueue(MQ_ROUDI_NAME);
-
-            m_rouDiMemoryManager.createAndAnnounceMemory().or_else([](RouDiMemoryManagerError error) {
-                LogFatal() << "Could not create SharedMemory! Error: " << static_cast<uint64_t>(error);
-                errorHandler(Error::kROUDI_COMPONENTS__SHARED_MEMORY_UNAVAILABLE, nullptr, iox::ErrorLevel::FATAL);
-            });
-            return true;
-        },
-        []() {}};
-
     /// @brief Handles the ports in shared memory
-    PortManager m_portManager{&m_rouDiMemoryManager};
+    PortManager m_portManager{initRouDiMemoryManager().value()};
```

Comment:
If `initRouDiMemoryManager` returns an error we run here directly into a segmentation fault since the expected is dereferencing a nullptr here.

Reply:
Done. TIL directly invoking lamdas is possbile =)